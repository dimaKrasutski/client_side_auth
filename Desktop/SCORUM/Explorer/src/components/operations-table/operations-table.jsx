import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FormattedMessage } from 'react-intl';
import Table from 'reactstrap/lib/Table';
import { Operation } from './operation/operation';
import { PreloaderOperations } from '../preloader/preloader-operations';

export const OperationsTableUI = ({
  operations,
  blockReady,
  locale,
  inProgress,
}) => (
  <Table className="mt-3 blocks-table">
    <thead className="thead-light">
      <tr>
        <th>
          <FormattedMessage id="explorer.block-info.transactions.hash" />
        </th>
        <th>
          <FormattedMessage id="explorer.block-info.transactions.date" />
        </th>
        <th>
          <FormattedMessage id="explorer.block-info.transactions.type" />
        </th>
      </tr>
    </thead>
    <tbody>
      {!operations.length && !blockReady ?
        <PreloaderOperations /> :
        operations.map((operation) => {
          const [opHash, opObj] = operation;
          return <Operation key={operation.trx_id} locale={locale} hash={opHash} operation={opObj} />;
        })
      }
      {
        inProgress && <PreloaderOperations />
      }
    </tbody>
  </Table>
);

OperationsTableUI.propTypes = {
  locale: PropTypes.string.isRequired,
  blockReady: PropTypes.bool.isRequired,
  inProgress: PropTypes.bool.isRequired,
  operations: PropTypes.arrayOf(
    PropTypes.shape({
      timestamp: PropTypes.string.isRequired,
      block_id: PropTypes.string.isRequired,
      height: PropTypes.number.isRequired,
      transactions: PropTypes.array.isRequired,
    }),
  ).isRequired,
};

export const mapStateToProps = ({ intl }) => ({
  locale: intl.localeFull,
});

export const OperationsTable = connect(mapStateToProps)(OperationsTableUI);
