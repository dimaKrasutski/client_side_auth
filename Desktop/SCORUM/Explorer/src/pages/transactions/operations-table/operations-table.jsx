import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import Table from 'reactstrap/lib/Table';
import { Transaction } from './transaction/transaction';
import { PreloaderOperations } from '../../../components/preloader/preloader-operations';

export const OperationsTable = ({ operations, blockReady, inProgress }) => (
  <Table className="mt-3 blocks-table">
    <thead className="thead-light">
      <tr>
        <th>
          <FormattedMessage id="explorer.block-info.transactions.hash" />
        </th>
        <th>
          <FormattedMessage id="explorer.block-info.transactions.date" />
        </th>
        <th>
          <FormattedMessage id="explorer.block-info.transactions.type" />
        </th>
      </tr>
    </thead>
    <tbody>
      {!operations.length && !blockReady ?
        <PreloaderOperations /> :
        operations.map((operation) => {
          const [, trxObj] = operation;
          return <Transaction key={trxObj.trx_id} transaction={trxObj} />;
        })
      }
      {
        inProgress && <PreloaderOperations />
      }
    </tbody>
  </Table>
);

OperationsTable.propTypes = {
  blockReady: PropTypes.bool.isRequired,
  inProgress: PropTypes.bool.isRequired,
  operations: PropTypes.arrayOf(
    PropTypes.shape({
      timestamp: PropTypes.string.isRequired,
      block_id: PropTypes.string.isRequired,
      height: PropTypes.number.isRequired,
      transactions: PropTypes.array.isRequired,
    }),
  ).isRequired,
};
