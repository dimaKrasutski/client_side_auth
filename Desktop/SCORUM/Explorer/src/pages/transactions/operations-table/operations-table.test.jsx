import React from 'react';
import { OperationsTable } from './operations-table';

describe('<OperationsTable />', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<OperationsTable operations={[]} blockReady />);
    expect(wrapper).toMatchSnapshot();
  });
});
