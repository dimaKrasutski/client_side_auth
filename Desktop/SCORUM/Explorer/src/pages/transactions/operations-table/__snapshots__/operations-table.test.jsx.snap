// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OperationsTable /> should match snapshot 1`] = `
<Table
  className="mt-3 blocks-table"
  responsiveTag="div"
  tag="table"
>
  <thead
    className="thead-light"
  >
    <tr>
      <th>
        <FormattedMessage
          id="explorer.block-info.transactions.hash"
          values={Object {}}
        />
      </th>
      <th>
        <FormattedMessage
          id="explorer.block-info.transactions.date"
          values={Object {}}
        />
      </th>
      <th>
        <FormattedMessage
          id="explorer.block-info.transactions.type"
          values={Object {}}
        />
      </th>
    </tr>
  </thead>
  <tbody />
</Table>
`;
